sentry/src/lib.rs

```diff
@@ -308,6 +324,28 @@ async fn channels_router<A: Adapter + 'static>(
     }
 }
 
+async fn campaigns_router<A: Adapter + 'static>(
+    mut req: Request<Body>,
+    app: &Application<A>,
+) -> Result<Response<Body>, ResponseError> {
+    req = AuthRequired.call(req, app).await?;
+
+    let (path, method) = (req.uri().path().to_owned(), req.method());
+
+    // regex matching for routes with params
+    if let (Some(caps), &Method::POST) = (CAMPAIGN_UPDATE_BY_ID.captures(&path), method) {
+        let param = RouteParams(vec![caps
```

Comment:
Here there are a few things we can make:
- [ ] Use the parameter and parse the `param.get(0).ok_or_else(BadRequest("No CampaignId")`
- [ ] Fetch `Campaign` using the parsed `CampaignId` ( None should be handled as ResponseError::NotFound)
- [ ] Add `Campaign` to `req.extensions_mut()` (see a few lines bellow)